services/device-registry-file/src/main/java/org/eclipse/hono/deviceregistry/file/FileBasedRegistrationService.java

```diff
@@ -538,12 +551,33 @@ public String toString() {
      */
     private String generateDeviceId(final String tenantId) {
 
-        final ConcurrentMap<String, Versioned<Device>> devices = getDevicesForTenant(tenantId);
+        final ConcurrentMap<String, FileBasedDeviceDto> devices = getDevicesForTenant(tenantId);
         String tempDeviceId;
         do {
             tempDeviceId = UUID.randomUUID().toString();
         } while (devices.containsKey(tempDeviceId));
         return tempDeviceId;
     }
 
+    private static final class FileBasedDeviceDto extends BaseDto<Versioned<Device>> {
```

Comment:
Can this be a more generic class, that could probably be reused for other implementation? I don't see anything specific to File implementation.

Reply:
Actually there is a small, specific part: the call to `getValue()` in `final DeviceWithStatus deviceWithStatus = new DeviceWithStatus(getData().getValue());`

The base class `BaseDto` from which this class is derived was envisioned as the generic class which is reused by other implementations.